### PR TITLE
Libretro: refactor paths

### PR DESCRIPTION
**Pull request by SimpleTease**

### CODE REVIEW

**Comment by frranck** [comment]
some stuff into libretro-common are also beeing used by the SDL versions I think.

```
air:mrboom-libretro fr$ make mrboom LIBSDL2=1
make: *** No rule to make target `libretro/libretro-common/file/file_path.o', needed by `mrboom'.  Stop.
```
Yep, so we should not move libretro-common, on the contrary the jni directory is pure retrolib

**Comment by frranck** [comment]
actually better _not_ move the jni, it's some android crap...

**Comment by SimpleTease** [comment]
I'm wondering.
https://github.com/Javanaise/mrboom-libretro/blob/3bdbc0cc127dcf0c55e984c7b7fccea20c629575/Makefile.common#L44-L48

Should this be?
```
 ifeq ($(LIBSDL2),) 
 ifeq ($(LIBSDL),) 
```

Although I see this later. So libretro-common should be on root level.
https://github.com/Javanaise/mrboom-libretro/blob/3bdbc0cc127dcf0c55e984c7b7fccea20c629575/Makefile.common#L85-L99

File: Makefile.common [hunk 1]
```diff
@@ -1,109 +1,113 @@
...
 ifeq ($(platform), ctr)
-SOURCES_C +=   $(CORE_DIR)/audio/retro_audio_mixer.c
+SOURCES_C +=    $(CORE_DIR)/libretro/audio/audio_mixer.c
 endif
 ifeq ($(platform), psl1ght)
-SOURCES_C +=   $(CORE_DIR)/audio/retro_audio_mixer.c
+SOURCES_C +=    $(CORE_DIR)/libretro/audio/audio_mixer.c
 endif
 else
+ifeq ($(LIBSDL2),)
```
Comment (frranck) [inline Makefile.common:47]: this is used by SDL2 , that's why the LIBRETRO_COMM_DIR should stay at the top

**Comment by SimpleTease** [comment]
Should `retro_data.h` and `retro_music_data.h` be moved into libretro also? Like sdl1/2?

File: Makefile.common (hunk 1 above)
Comment (SimpleTease) [reply Makefile.common]: I'm slow, but I get it now.

**Comment by frranck** [comment]
No they're both for libretro:

```
#ifdef __LIBRETRO__
#include <audio/audio_mixer.h>
#include <audio/conversion/float_to_s16.h>
static float *              fbuf = NULL;
static int16_t *            ibuf = NULL;
static audio_mixer_sound_t *musics[NB_CHIPTUNES];
#ifndef LOAD_FROM_FILES
#include "retro_data.h"
#include "retro_music_data.h"
#endif
```
I had a second thought, maybe you could try jni too. there are some makefiles in there and you should fix the .gitlab-ci,yml

**Comment by SimpleTease** [comment]
I moved retro_* into libretro folder and it compiles.

> I had a second thought, maybe you could try jni too. there are some makefiles in there and you should fix the .gitlab-ci,yml

I think you should ask @twinaphex. I'm not familiar with the internals of buildbot. Libretro has some rules sometimes.

**Comment by frranck** [comment]
OK no worries, should we merge? did you check the SDL2 is still OK?

**Comment by SimpleTease** [comment]
It's safer if you check SDL2. It should be good. But. I have different experiences sometimes, so I can't offer fail-proof guarantee.

**Comment by frranck** [comment]
It's easy you need to make clean first though, then make mrboom LIBSDL2=1

**Comment by SimpleTease** [comment]
I think `jni` can be moved around safely. I can compile Android so I'll follow-up.
https://github.com/libretro/desmume/blob/master/.gitlab-ci.yml
https://github.com/libretro/desmume/pull/37/commits/2c72cd037e4907fbeff9cfcd3e2a00a09b6c177e

**Pull request merged**